dissect/ffs/ffs.py

```diff
@@ -259,6 +266,10 @@ def iterdir(self):
 
         while offset < self.size - 8:
             dirent = c_ffs.direct(buf)
+            if dirent.d_reclen == 0:
```

Comment:
I think it would be good if we have a unit test that demonstrates how this change prevents an infinite loop? This also makes sure this issue will never return again...

Reply:
If you can help with a mock setup for this, that would be appreciated :)

Reply:
Sure, no problem, I will look into this.

Reply:
Something like this?


```
@patch("dissect.ffs.ffs.INode.open")
@patch("dissect.ffs.ffs.INode.__repr__", return_value="<dissect.ffs.ffs.INode>")
@patch("dissect.ffs.ffs.INode.size", create=True, return_value=13)
@patch("dissect.ffs.ffs.FFS")
@patch("dissect.ffs.ffs.log", create=True, return_value=None)
@patch("dissect.ffs.ffs.c_ffs.direct", create=True)
def test_infinite_loop_protection(direct, log, FFS, *args):
    class Dummy:
        def __init__(self,data):
            self.__dict__ = data
    direct.return_value = Dummy({
        "d_reclen": 0,
        "d_namlen": 1,
        "d_type": 1,
        "d_ino":1
    })
    fsinode = Dummy({"di_size":10, "inum":1,"sb":"","inode":FFS})
    inode = INode(fsinode, FFS(1), 1, filetype=stat.S_IFDIR)
    inode.size = 13
    for entry in inode.iterdir():
        pass
    assert call.critical('Zero-length directory entry in %s (offset 0x%x)', inode, 0) in log.mock_calls
```